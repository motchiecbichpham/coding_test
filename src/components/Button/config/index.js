export const DEFAULT_SIZE_ACTIVITY_INDICATOR = 'small';
