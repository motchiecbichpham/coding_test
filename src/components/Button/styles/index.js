import {StyleSheet} from 'react-native';
import {COLORS} from '../../../utils/colors';
const ButtonStyles = StyleSheet.create({
  button: {
    height: 48,
    width: '70%',
    paddingHorizontal: 16,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 8,
    marginTop: 16,
    backgroundColor: COLORS.grayAsparagus,
  },
  buttonText: {
    fontWeight: 'bold',
    fontSize: 16,
    color: COLORS.eggshell,
  },
});

export default ButtonStyles;
