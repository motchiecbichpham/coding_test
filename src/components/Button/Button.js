//import libraries
import React from 'react';
import {ActivityIndicator, Text, TouchableOpacity} from 'react-native';
import ButtonStyles from './styles/index';
import {DEFAULT_SIZE_ACTIVITY_INDICATOR} from './config';
import {COLORS} from '../../utils/colors';
const Button = ({title, textColor, style, onPress, disabled, isLoading}) => {
  if (isLoading) {
    onPress = null;
  }
  return (
    <TouchableOpacity
      activeOpacity={0.7}
      disabled={disabled}
      style={[ButtonStyles.button, style]}
      onPress={onPress}>
      {isLoading ? (
        <ActivityIndicator
          size={DEFAULT_SIZE_ACTIVITY_INDICATOR}
          color={textColor || COLORS.white}
        />
      ) : (
        <Text style={ButtonStyles.buttonText}>{title}</Text>
      )}
    </TouchableOpacity>
  );
};
export default Button;
