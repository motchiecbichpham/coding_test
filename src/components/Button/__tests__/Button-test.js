import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

const buttonProps = {
  title: 'button',
  textColor: '#333',
  disable: false,
  isLoading: false,
  onPress: () => {},
};
test('render button', () => {
  const tree = renderer.create(<Button {...buttonProps} />).toJSON();
  expect(tree).toMatchSnapshot();
});
