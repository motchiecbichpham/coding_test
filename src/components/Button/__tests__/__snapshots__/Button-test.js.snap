// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#3A6351",
      "borderRadius": 8,
      "flexDirection": "row",
      "height": 48,
      "justifyContent": "center",
      "marginTop": 16,
      "opacity": 1,
      "paddingHorizontal": 16,
      "width": "70%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#F2EDD7",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    button
  </Text>
</View>
`;
